libportability-gfx/src/conv.rs

```diff
@@ -720,3 +720,43 @@ pub fn map_pipeline_statistics(flags: VkQueryPipelineStatisticFlags) -> query::P
     // Vulkan and HAL flags are equal
     unsafe { mem::transmute(flags) }
 }
+
+pub fn map_specialization_info(specialization: &VkSpecializationInfo) -> Vec<pso::Specialization> {
+    let data = unsafe { slice::from_raw_parts(
+        specialization.pData as *const u8,
+        specialization.dataSize as _,
+    )};
+    let entries = unsafe { slice::from_raw_parts(
+        specialization.pMapEntries,
+        specialization.mapEntryCount as _,
+    )};
+
+    entries
+        .into_iter()
+        .map(|entry| {
+            let offset = entry.offset as usize;
+            pso::Specialization {
+                id: entry.constantID,
+                value: match entry.size {
+                    4 => pso::Constant::U32(
+                        data[offset] as u32 |
```

Comment:
nit: could probably look cleaner with a `fold`

Reply:
@msiglreith would probably argue that it doesn't worth the risk of overhead... and that would be correct.
I was thinking about something like this:
```rust
data[offset..offset+4].iter().rev().fold(0u32, |u, &v| (u << 8) | v)
```

Reply:
Ah ok, so should I just leave this as it is for now?